datafusion/physical-expr/src/aggregate/utils.rs

```diff
@@ -31,3 +33,49 @@ pub fn get_accum_scalar_values_as_arrays(
         .map(|s| s.to_array_of_size(1))
         .collect::<Vec<_>>())
 }
+
+pub fn calculate_result_decimal_for_avg(
```

Comment:
This seems somewhat related to https://github.com/apache/arrow-datafusion/pull/5675 from @viirya  to calculate the output size of a decimal multiplication. I wonder if there is some more general function / approach than special casing in the aggregator

Reply:
I think it is a little bit different then the explicit arithmetic expression in the SQL,  the reason is this final result decimal conversion is implicit, the logical plans/optimizers actually are not aware of this implicit conversion. This is similar to the implicit `cast` that I pulled up this PR, the cast is also implicit and the logical plans/optimizers is not aware of this implicit `cast` either. Those implicit `cast` and conversion is only related to internal state of some specific Aggregation `Accumulator`, that are not part of the SQL plans/expressions tree.